src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,96 @@
 package core.basesyntax;
 
 public class SalaryInfo {
+    private static final int RECORD_DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_STR_INDEX = 2;
+    private static final int RATE_STR_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        return generateReport(names, data, dateFrom, dateTo);
+    }
+
+    private String generateReport(String[] names, String[] data, String dateFrom, String dateTo) {
+        StringBuilder reportBuilder = new StringBuilder();
+
+        initializeReportHeader(reportBuilder, dateFrom, dateTo);
+        int[] earnings = calculateEarnings(names, data, dateFrom, dateTo);
+        finalizeReport(reportBuilder, names, earnings);
+
+        return reportBuilder.toString();
+    }
+
+    private void initializeReportHeader(StringBuilder reportBuilder,
+                                        String dateFrom,
+                                        String dateTo) {
+        reportBuilder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+    }
+
+    private int[] calculateEarnings(String[] names, String[] data, String dateFrom, String dateTo) {
+
+        int[] earnings = new int[names.length];
+
+        for (String record : data) {
+            String[] parts = record.split(" ");
+            if (parts.length == 4) {
+                String recordDate = parts[RECORD_DATE_INDEX];
```

Comment:
You don't need to store this values as strings - parse them immediately:
- `LocalDate recordDate`
- `int hours`
- `int rate`